Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperationResult.swift

```diff
@@ -120,3 +154,128 @@ extension DeleteResult: BulkWriteResultConvertible {
         BulkWriteResult.new(deletedCount: self.deletedCount)
     }
 }
+
+struct ErrorResult: Equatable, Matchable {
+    internal static let errorKeys: Set = ["errorContains", "errorCodeName", "errorLabelsContain", "errorLabelsOmit"]
+
+    internal var errorContains: String?
+
+    internal var errorCodeName: String?
+
+    internal var errorLabelsContain: [String]?
+
+    internal var errorLabelsOmit: [String]?
+
+    public init(from doc: Document) {
```

Comment:
I think it would be a little more consistent if we still relied on `Decodable` to generate this struct rather than doing it ourselves. To avoid collisions with the `document` case, we can check that at least one of the keys is present via the `CodingKey`s, and throw an error if none are. We currently do that for [`BulkWriteResult`](https://github.com/mongodb/mongo-swift-driver/blob/master/Sources/MongoSwift/MongoCollection%2BBulkWrite.swift#L299).